packages/custom_lint/lib/src/package_utils.dart

```diff
@@ -77,7 +77,13 @@ Future<Pubspec?> tryParsePubspec(Directory directory) async {
 ///
 /// Throws if the parsing fails, such as if the file is badly formatted or
 /// does not exists.
-Future<Pubspec> parsePubspec(Directory directory) async {
+Future<Pubspec> parsePubspec(Directory directory, [Directory? original]) async {
+  if (!directory.pubspec.existsSync()) {
+    if (directory.parent.path == directory.path) {
```

Comment:
Could we straight up do the following?

```suggestion
    if (directory.parent == directory) {
```

Reply:
Directory doesn't override (==), but Uri does, so I'll switch to that